notebooks/02-ingestion/04-table-datatypes.ipynb

```diff
@@ -0,0 +1,851 @@
+    "  TIME_FLOOR(CAST(\"__time_arrival\" AS TIMESTAMP),'P1D') AS \"day-2\",\n",
+    "  AVG(TIMESTAMPDIFF(SECOND, \"__time\", NVL(MILLIS_TO_TIMESTAMP(\"__time_arrival\"),\"__time\"))) AS \"averageFlightLength_s\"\n",
+    "FROM \"example-flights-types-2\"\n",
+    "WHERE TIME_IN_INTERVAL(\"__time\",'2005-11-01/P7D')\n",
+    "GROUP BY 1, 2\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c99b46a5-3d31-410a-849f-31f67dc7ec8b",
+   "metadata": {},
+   "source": [
+    "When wanting to parse string data as a timestamp, there are a variety of [functions](https://druid.apache.org/docs/latest/querying/sql-scalar#date-and-time-functions) that support different formats.\n",
+    "\n",
+    "Run the cell below to see CAST being used to parse a SQL-standard datetime as a TIMESTAMP.\n",
+    "\n",
+    "> For more examples, see the dedicated notebook on [time functions](../03-query/07-functions-datetime.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2faf4fd3-70ce-4ff5-a83f-92c90f22617d",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT CAST('2000-01-02 03:04:05' AS TIMESTAMP) AS \"castFromSqlString\"\n",
+    "FROM \"example-flights-types-2\"\n",
+    "LIMIT 1\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1e8f9703-1653-4687-a506-99a52f209642",
+   "metadata": {},
+   "source": [
+    "### Booleans\n",
+    "\n",
+    "The result of the boolean CASE function is also a BIGINT.\n",
+    "\n",
+    "```sql\n",
+    "    CASE WHEN \"Origin\" = 'SFO' THEN true ELSE false END AS \"isSilicon\",\n",
+    "```\n",
+    "\n",
+    "Run the cell below to see what this looks like in the data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "08d7fa41-7647-4624-a226-533982840dd9",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sql='''\n",
+    "SELECT\n",
+    "  \"isSilicon\",\n",
+    "  COUNT(*) AS \"flights\"\n",
+    "FROM \"example-flights-types-2\"\n",
+    "WHERE TIME_IN_INTERVAL(\"__time\",'2005-11-08/P7D')\n",
+    "GROUP BY 1\n",
+    "'''\n",
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f726077e-b10f-4cf2-af47-df9ca89d9cc9",
+   "metadata": {},
+   "source": [
+    "Conversion is required to ensure Druid applies the configured [strict boolean behaviour](https://druid.apache.org/docs/latest/querying/sql-data-types#boolean-logic), [vectorization](https://druid.apache.org/docs/latest/misc/math-expr/#vectorization-support), and approach to [logical operations](https://druid.apache.org/docs/latest/misc/math-expr/#logical-operator-modes) - particulary important when controlling how NULLs are handled.\n",
+    "\n",
+    "Run the following cell to see examples of CAST converts both `isSilicon` and other numeric data in the table to boolean."
```

Comment:
I would add somewhere that any number can be cast to BOOLEAN but that only "0" will result in False, all other values are considered True.

Reply:
Raised a question with the internal team as I got some strange results.
Also expanded the examples.